solutions/usaco-717.mdx

```diff
@@ -2,14 +2,93 @@
 id: usaco-717
 source: USACO Gold 2017 December
 title: Why Did the Cow Cross the Road
-author: Qi Wang
+author: Qi Wang, Jeffrey Zhang
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_visitfj_gold_feb17.html)
 
+## Solution
+
+Instead of treating this problem with grids, let us consider from a more graphical perspective with nodes and edges.
+
+We first observe that because Bessie only feasts on grass tiles every 3 moves and the time to travel between adjacent tiles is a constant $T$. The two intermediary tiles Bessie used to travel does not actually matter. So we want to form a direct edge between each tile that are exactly 3 manhattan distances apart, with an weight of 
+
+$\{\text{The time to feast the target grass tile}\} + 3 * T$
+
+However, if the tile Bessie just feasted on is less than 3 manhattan distances away from Farmer John's tile, then it would be more desirable for Bessie to travel to Farmer John directly than take another 3-distanced tile. So let us too form a direct edge between those two tiles with an weight of 
```

Comment:
```suggestion
However, if the tile Bessie just feasted on is less than 3 units (using manhattan distance) away from Farmer John's tile, then it would be more desirable for Bessie to travel to Farmer John directly than take another 3-distanced tile. So, we also form a direct edge between those two tiles with a weight of 
```